### PR TITLE
Patch and test some edge cases for deserializing OneDrive metadata

### PR DESCRIPTION
**Pull request by ashmrtn**
## Description

Currently no folder path entry is stored for the root folder. This leads to not having a folders map but having a valid delta token if all items are in the root of the drive.

Update the code to add at least an empty folders map entry so we don't think we have missing metadata. Add tests to check for these edge cases.

Long-term we may want to add a folder entry for the root folder. Whether we do or not may depend on how we decide to set the state field for that collection.

## Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No 

## Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

## Issue(s)

* #2122 

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by ashmrtn** [comment]
@meain tagging you here as there's some stuff about folder path metadata you may need to change when setting state fields

(from PR description)
>Long-term we may want to add a folder entry for the root folder. Whether we do or not may depend on how we decide to set the state field for that collection.

File: src/internal/connector/onedrive/collections.go [hunk 1]
```diff
@@ -180,7 +180,15 @@ func deserializeMetadata(
 
 		// Go through and remove partial results (i.e. path mapping but no delta URL
 		// or vice-versa).
-		for k := range prevDeltas {
+		for k, v := range prevDeltas {
+			// Remove entries with an empty delta token as it's not useful.
+			if len(v) == 0 {
```
Comment (meain) [inline src/internal/connector/onedrive/collections.go:185]: Just wondering what is the scenario where this would happen.

**Review by meain — APPROVED** [review]

File: src/internal/connector/onedrive/collections.go (hunk 1 above)
Comment (ashmrtn) [reply src/internal/connector/onedrive/collections.go:185]: I don't expect this to happen at the moment, but I do want to make sure we handle it if it does. Right now the code that serializes the delta tokens has it's own check to ensure no 0 length tokens are added to the metadata. However, having a check here makes the code a little more flexible because even if the other check is accidentally removed things should (hopefully) continue working as expected
